### DISCUSSION TITLE
Using buttons as checkboxes to create an array of choices

### DISCUSSION THREAD

**Discussion by jvgeee** [react-hook-form/react-hook-form#4484] [Q&A] [answered]
Can't find a good answer online as to whether what I'm trying to do is a "field array" or not, but that's a separate problem!

I'm using a Controller for a wizard-style form (in React Native, but that shouldn't affect anything here). I want to have buttons that act as checkboxes, and create form output of:

```
  { fieldName: ["Choice 1", "Choice 5"] }
```

I can get part way there by using `name={`fieldName[${index}]`}`, but this gives me an output of:

```
 { fieldName: ["Choice 1", null, null, null, "Choice 5"] }
```
Any way to avoid nulls/empty strings etc? [Here's a code sandbox](https://codesandbox.io/s/react-hook-form-array-fields-forked-vhqkh?file=/src/index.js), and the relevant code:

```
return (
    <div>
      {activeOptions.map((item, key) => {
        const text = item.label;
        const isSelected = activeAnswers?.includes(text);

        return (
          <Controller
            key={key}
            control={control}
            render={({ onChange, onBlur, value }) => (
              <button
                preset={isSelected ? "outline_selected" : "outline"}
                onClick={() => onChange(isSelected ? null : text)}
              >
                {text}
              </button>
            )}
            name={`${activeFieldName}[${key}]`}
            defaultValue={null}
          />
        );
      })}
      <h5 style={{ color: "white" }}>{JSON.stringify(activeAnswers)}</h5>
    </div>
  );
```

**Comment by bluebill1049** [comment, marked as answer]
I think it would be easier to build a group component and wrapped with `Controller`, you can sort out the internal input state (group checkbox/radio) inside that component. Controller will only be the guy to collect value and does the validation.

**Reply by jvgeee** [reply]
Interesting. I've edited the code sandbox to reflect this: https://codesandbox.io/s/react-hook-form-array-fields-forked-vhqkh?file=/src/index.js

Seems to be working fine this way, please let me know if you see me doing anything majorly wrong here! Cheers @bluebill1049 .

**Reply by bluebill1049** [reply]
awesome! looks much better. `Controller` is just a "collector" and checks a value/values and validate the result, so this way it's much more efficient and cleaner. Good job.
